widget/src/button.rs

```diff
@@ -505,6 +508,47 @@ impl Default for Style {
 }
 
 /// The theme catalog of a [`Button`].
+/// 
+/// All themes that can be used with [`Button`]
+/// must implement this trait.
+/// 
+/// # Example
+/// ```no_run
+/// #[derive(Debug, Default)]
+/// pub enum ButtonStyles {
+///     #[default]
+///     Primary,
+///     Secondary,
+///     Danger
+/// }
+/// 
+/// impl Catalog for MyTheme {
+///     type Class<'a> = ButtonStyles;
+///     
+///     fn default<'a>() -> Self::Class<'a> {
+///         ButtonStyles::default()
+///     }
+///     
+/// 
+///     fn style(&self, class: &Self::Class<'_>, status: Status) -> Style {
+///         let mut style = Style::default();
+/// 
+///         match class {
+///             ButtonStyles::Primary => {
+///                 style.background = Some(Background::Color(Color::from_rgb(0.529, 0.808, 0.921)));
+///             },
+///             ButtonStyles::Secondary => {
+///                 style.background = Some(Background::Color(Color::WHITE));
+///             },
+///             ButtonStyles::Danger => {
+///                 style.background = Some(Background::Color(Color::from_rgb(0.941, 0.502, 0.502)));
+///             },
+///         }
+/// 
+///         style
+///     }
+/// }
+/// ```
 pub trait Catalog {
     /// The item class of the [`Catalog`].
```

Comment:
May be interesting to mention here in the `Class` that `Button::style` can only be used if the associated type here implements: `From<StyleFn<'a, Self>>`.

Reply:
Done